### PR TITLE
Energy electron volt

### PR DESCRIPTION
**Pull request by cquiroz**
I noted that we are missing the very important `Electron-Volt` unit for energy. This PR adds them

I'll probably add the other uses of `eV` for `Mass`, `Momentum`,  `Distance` and `Temperature`

Do you think I should add a `UnitGroup` for these?

### CODE REVIEW

**Comment by derekmorr** [comment]
We can't add a unit group for them since they're different Dimensions.

**Comment by cquiroz** [comment]
@derekmorr Wouldn't e.g. `eV`, `keV`, `MeV` belong to the same group?

**Comment by derekmorr** [comment]
Oh, sorry. I misunderstood. I thought you meant adding units for `Mass`, `Momentum`, etc into a unit group.

I agree that naming will be an issue.

File: shared/src/main/scala/squants/energy/Energy.scala [hunk 1]
```diff
@@ -210,6 +261,16 @@ object EnergyConversions {
...
 
+  lazy val eV = ElectronVolt(1)
+  lazy val meV = MilliElectronVolt(1)
+  lazy val keV = KiloElectronVolt(1)
+  lazy val MeV = MegaElectronVolt(1)
+  lazy val GeV = GigaElectronVolt(1)
+  lazy val TeV = TeraElectronVolt(1)
+  lazy val PeV = PetaElectronVolt(1)
+  lazy val EeV = ExaElectronVolt(1)
+  lazy val evMultiplier: Double = 1.60217656535e-19
```
Comment (derekmorr) [inline shared/src/main/scala/squants/energy/Energy.scala]: Should this be moved into the `ElectronVolt` object so we don't leak a constant?

Comment (cquiroz) [inline shared/src/main/scala/squants/energy/Energy.scala]: Yes, that's probably a good idea

**Comment by derekmorr** [comment]
This LGTM. Did we decide on how to handle naming issues?

**Comment by cquiroz** [comment]
No decision yet, but I think it will be a problem to have the same unit name. I'm leaning to use `ElectronVolt` only for `Energy` and add a prefix in the other cases like `ElectronVoltMass`, `ElectronVoltTemperature`, etc

WDYT?

File: shared/src/main/scala/squants/energy/Energy.scala [hunk 2]
```diff
@@ -187,6 +198,46 @@ object Ergs extends EnergyUnit {
   val symbol = "erg"
 }
 
+object ElectronVolt extends EnergyUnit {
+  val conversionFactor = Joules.conversionFactor * 1.60217656535e-19
```
Comment (cquiroz) [inline shared/src/main/scala/squants/energy/Energy.scala]: Thanks for poiting this up, I'll update this and the constants PR that uses the same bad logic

**Comment by cquiroz** [comment]
Seems I could merge this. I'll wait a bit to let others chime in if needed

**Pull request merged**